### PR TITLE
Bridge vlan filtering

### LINKED ISSUE #113: Support bridge vlan filtering
MikroTik documentation: https://wiki.mikrotik.com/wiki/Manual:Interface/Bridge#Bridge_VLAN_Filtering
Sub-menu: `/interface bridge vlan`

### LINKED ISSUE #114: Support bridge vlan filtering
Is blocked by #113

### PR DESCRIPTION
**Pull request by maksym-nazarenko**
Closes #113 , closes #114 

This PR contains:
* bridge VLAN support (client and resource)
* new `MikrotikIntList` type for `client`
* fix for `dhcp_lease` when returned `nil` object was accessed before error check

### CODE REVIEW

File: client/bridge_vlan.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+// BridgeVlan defines vlan filtering in bridge resource
+type BridgeVlan struct {
+	Id       string                `mikrotik:".id"`
+	Bridge   string                `mikrotik:"bridge"`
+	Tagged   types.MikrotikList    `mikrotik:"tagged"`
+	Untagged types.MikrotikList    `mikrotik:"untagged"`
+	VlanIds  types.MikrotikIntList `mikrotik:"vlan-ids"`
+}
+
+var _ Resource = (*BridgeVlan)(nil)
```
Comment (ddelnano) [inline client/bridge_vlan.go:17]: What does this accomplish? I cloned your branch locally and everything compiles and runs correctly without it.

**Comment by ddelnano** [comment]
Just one comment about the purpose of the `Resource` variable created in [client/bridge_vlan.go](https://github.com/ddelnano/terraform-provider-mikrotik/pull/123/files/291a6b6df584fb50e1cf7fe383f3709a5bb02026#diff-a39f56a210541cc0cdd0ff2c4c06ace0697ffbc02f502fac6e5ed3c1e069a49b). Otherwise this lgtm.

File: client/bridge_vlan.go (hunk 1 above)
Comment (maksym-nazarenko) [reply client/bridge_vlan.go:17]: It is a way to force interface implementation, a reminder in case `Resource` interface changes but you forget to update the struct. If that happens, you will get an error on this line that `cannot use *BridgeVlan as Resource: missing method ...` (you can check it by simple commenting `func (b *BridgeVlan) IDField() string {}` function)
Also, it is useful if you have optional interface like [`Finder`](https://github.com/ddelnano/terraform-provider-mikrotik/blob/374cae8383e8fd53ad50ce58ac226147a1bee561/client/client_crud.go#L47) which also could be tested with `var _ Finder = (*BridgeVlan)(nil)`


Anyway, it is a defensive tactic, we can safely remove it. If we have problems in the future, we can bring it back.
Please, feel free to remove

Comment (ddelnano) [reply client/bridge_vlan.go:17]: ah, I see now. Using it sounds good to me, I just wanted to make sure I understood its purpose.
